server/data/data-jmap/src/test/java/org/apache/james/jmap/api/preview/MessagePreviewStoreContract.java

```diff
@@ -119,25 +120,37 @@ default void storeShouldBeIdempotent() {
     }
 
     @Test
-    default void concurrentStoreShouldStoreAllRecords() throws Exception {
+    default void concurrentStoreShouldStoreTheLatestValueDespiteOfCollisions() throws Exception {
         int threadCount = 10;
         int stepCount = 100;
-
-        ConcurrentTestRunner.builder()
-            .operation((thread, step) -> {
-                int id = thread * stepCount + step;
-                Mono.from(testee().store(TestMessageId.of(id), Preview.from(String.valueOf(id))))
-                    .block();
+        int repeatCount = 10;
+
+        ConcurrentHashMap<Integer, Integer> indexCounter = new ConcurrentHashMap<>();
+
+        Flux.range(0, stepCount)
+            .repeat(repeatCount)
+            .parallel(threadCount)
+            .runOn(Schedulers.newParallel("preview", threadCount))
+            .flatMap(index -> {
+                Integer counter = indexCounter.compute(index, (id, count) -> {
```

Comment:
this part of the code probably won't run where you think, you need to wrap it in a Mono